content/sensu-go/6.3/operations/manage-business-services/rule-templates.md

```diff
@@ -0,0 +1,472 @@
+{{< /code >}}
+{{< code json >}}
+{
+  "description": "Creates an event when the percentage of events with the given status exceed the given threshold"
+}
+{{< /code >}}
+{{< /language-toggle >}}
+
+arguments    | 
+-------------|------ 
+description  | The rule template's arguments using [JSON Schema][1] properties.
+required     | true
+type         | Map of key-value pairs
+example      | {{< language-toggle >}}
+{{< code yml >}}
+arguments:
+  required:
+    - threshold
+  properties:
+    status:
+      type: string
+      default: non-zero
+      enum:
+        - non-zero
+        - warning
+        - critical
+        - unknown
+    threshold:
+      type: number
+      description: Numeric value that triggers an event when surpassed
+{{< /code >}}
+{{< code json >}}
+{
+  "arguments": {
+    "required": [
+      "threshold"
+    ],
+    "properties": {
+      "status": {
+        "type": "string",
+        "default": "non-zero",
+        "enum": [
+          "non-zero",
+          "warning",
+          "critical",
+          "unknown"
+        ]
+      },
+      "threshold": {
+        "type": "number",
+        "description": "Numeric value that triggers an event when surpassed"
+      }
+    }
+  }
+}
+{{< /code >}}
+{{< /language-toggle >}}
+
+eval         | 
+-------------|------
+description  | ECMAScript 5 (JavaScript) expression for the rule template to evaluate.
+required     | true
+type         | String
+example      | {{< language-toggle >}}
+{{< code yml >}}
+eval: |
+    var statusMap = {
+      "non-zero": 1,
+      "warning": 1,
+      "critical": 2,
+    };
+    function main(args) {
+      var total = sensu.events.count();
+      var num = sensu.events.count(args.status);
+      if (num / total <= args.threshold) {
+        return;
+      }
+      return sensu.new_event({
+        status: statusMap[args.status],
+        /* ... */,
+      });
+    }
+{{< /code >}}
+{{< code json >}}
+{
+  "eval": "var statusMap = {\n  \"non-zero\": 1,\n  \"warning\": 1,\n  \"critical\": 2,\n};\nfunction main(args) {\n  var total = sensu.events.count();\n  var num = sensu.events.count(args.status);\n  if (num / total <= args.threshold) {\n    return;\n  }\n  return sensu.new_event({\n    status: statusMap[args.status],\n    /* ... */,\n  });\n}"
+}
+{{< /code >}}
+{{< /language-toggle >}}
+
+#### Arguments attributes
+
+required     | 
+-------------|------ 
+description  | List of keys the rule template argument requires. The listed keys must be configured in the [properties][2] object.
+required     | true
+type         | Array
+example      | {{< language-toggle >}}
+{{< code yml >}}
+required:
+  - threshold
+{{< /code >}}
+{{< code json >}}
+{
+  "required": [
+    "threshold"
+  ]
+}
+{{< /code >}}
+{{< /language-toggle >}}
+
+<a name="rule-properties"></a>
+
+properties   | 
+-------------|------ 
+description  | Lists of properties that define the rule template's behavior. In [JSON Schema][1].
```

Comment:
Another guess -- I'm not sure what to list for the description.

Reply:
Also, these properties may be different depending on what you want the rule template to do, is that correct?